shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1696,116 +1679,121 @@ - (void)startKeyBoardAnimation:(NSTimeInterval)duration {
 
   // When this method is called for the first time,
   // initialize the keyboardAnimationView to get animation interpolation during animation.
-  if ([self keyboardAnimationView] == nil) {
+  if (!self.keyboardAnimationView) {
     UIView* keyboardAnimationView = [[UIView alloc] init];
-    [keyboardAnimationView setHidden:YES];
-    _keyboardAnimationView.reset(keyboardAnimationView);
+    keyboardAnimationView.hidden = YES;
+    self.keyboardAnimationView = keyboardAnimationView;
   }
 
-  if ([self keyboardAnimationView].superview == nil) {
-    [self.view addSubview:[self keyboardAnimationView]];
+  if (!self.keyboardAnimationView.superview) {
+    [self.view addSubview:self.keyboardAnimationView];
   }
 
   // Remove running animation when start another animation.
-  [[self keyboardAnimationView].layer removeAllAnimations];
+  [self.keyboardAnimationView.layer removeAllAnimations];
 
   // Set animation begin value and DisplayLink tracking values.
-  [self keyboardAnimationView].frame =
+  self.keyboardAnimationView.frame =
       CGRectMake(0, _viewportMetrics.physical_view_inset_bottom, 0, 0);
   self.keyboardAnimationStartTime = fml::TimePoint().Now();
   self.originalViewInsetBottom = _viewportMetrics.physical_view_inset_bottom;
 
   // Invalidate old vsync client if old animation is not completed.
   [self invalidateKeyboardAnimationVSyncClient];
 
-  fml::WeakNSObject<FlutterViewController> weakSelf = [self getWeakNSObject];
-  FlutterKeyboardAnimationCallback keyboardAnimationCallback = ^(
-      fml::TimePoint keyboardAnimationTargetTime) {
-    if (!weakSelf) {
-      return;
-    }
-    fml::scoped_nsobject<FlutterViewController> flutterViewController(
-        [(FlutterViewController*)weakSelf.get() retain]);
-    if (!flutterViewController) {
-      return;
-    }
+  __weak FlutterViewController* weakSelf = self;
+  FlutterKeyboardAnimationCallback keyboardAnimationCallback =
+      ^(fml::TimePoint keyboardAnimationTargetTime) {
+        FlutterViewController* strongSelf = weakSelf;
+        if (!strongSelf) {
```

Comment:
Optional: I highly recommend pulling all of the code in this block into a helper method. Once there's more than a very small amount of code, a helper is pretty much always better because you can just write normal code that uses `self` through all of the logic. And you don't even need the strong dance in the caller in that case, because you can just do make the whole method `[weakSelf myHelper];` and get the promotion to `strong` for free for the scope of the call.

Reply:
Good call. Callback. Regardless, done.